### DISCUSSION TITLE
What's the typical structure of someone using rocks.vim and how to load a custom package with it?

### DISCUSSION THREAD

**Discussion by Folaht** [lumen-oss/rocks.nvim#592] [Q&A] [answered]
I figured out how to install and prune packages with rocks.nvim, 
but I'm coming from Lunarvim and I don't know how to get these packages to work.  
I've figured out colorscheme packages so far, but I'm using vim for rust development,  
so I'm trying to figure out what Lunarvim was using for me, that would allow me to see
number lines, tabs, have code completion and display code errors.  

Not to mention that I want to use my custom made orthogonal dvorak keymappings again, 
because jhkl or whatever qwerty has for left/right up/down doesn't work for me,
and I prefer having all the similar commands on the same row 
like mwvz for delete(cut)/yank(copy)/paste/undo.   

```
~/.config/nvim
├── init.lua
├── keymaps_for_dvorak.lua
└── rocks.toml
```

rocks.toml
```
...
[rocks]

# List of Neovim plugins to install alongside their versions.
# If the plugin name contains a dot then you must add quotes to the key name!
[plugins]
"rocks.nvim" = "2.42.2"
"tokyonight.nvim" = "4.10.0"
...
"mason.nvim" = "1.10.0"
"mason-lspconfig.nvim" = "1.31.0"
```

init.lua
```
local rocks_config = {
    rocks_path = vim.fn.stdpath("data") .. "/rocks",
}
...
vim.opt.runtimepath:append(vim.fs.joinpath(rocks_config.rocks_path, "lib", "luarocks", "rocks-5.1", "rocks.nvim", "*"))

vim.cmd.colorscheme('tokyonight')

-- require('keymaps_for_dvorak')
```

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

We don't have migration guides for Neovim distros yet, but perhaps you may find [this answer to a similar question](https://github.com/nvim-neorocks/rocks.nvim/discussions/139#discussioncomment-8429847) helpful.

**Comment by Folaht** [comment]
Alright, that quickly fixes the structure.  
I'm still unsure about getting Mason to work, 
but I'll post that in a different question.

```
~/.config/nvim
├── init.lua
├── plugin
│   ├── colorscheme.lua
│   └── keymappings
│       └── dvorak.lua
└── rocks.toml
```
